### PR TITLE
feat: basic test using Maestro + set up Maestro

### PR DESCRIPTION
**Pull request by alexandre-mrt**
End to end test that uses Maestro a tool that simulate a user utilisation. 

## Installation ##
We had to install it first (find informations here: https://maestro.mobile.dev/api-reference/) 

Dependencies are necessary, including an android simulator. We used Orbit Expo to simulate on Android. 

After building the APK of the app, testing with maestro is available running the following command:
maestro test test-maestro/registration-test.yaml (replace with the right path to the yaml test)
 

see screenshot here: 
image

## Contributions : ##
- Simulation of a simple login user flow in the login-test.yaml 
- Simulation of a registration user flow in the test-maestro/registration-test.yaml

I enable a recording instruction and the end result can be seen here: 

https://github.com/uniconnect-epfl/uniconnect/assets/108724823/045f6157-22ef-438e-8f48-b8d2bed0ac3e

### CODE REVIEW

**Comment by alexandre-mrt** [comment]
Unable to merge due to last minutes problem with the whole project.

File: eas.json [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    "preview2": {
+      "android": {
+        "gradleCommand": ":app:assembleRelease"
+      }
+    },
+    "preview3": {
+      "developmentClient": true
+    },
+    "preview4": {
+      "distribution": "internal"
```
Comment (Aidasvenc) [inline eas.json:20]: No need to have 4 preview build profiles. The two that we will need are `development` and `production` - one for us to test the app and one for the releases and milestones.

File: screens/Onboarding/OnboardingScreen.tsx [hunk 2]
```diff
@@ -24,6 +24,7 @@ const OnboardingScreen: React.FC = () => {
       {/* Username/Email Field */}
       <TextInput
         style={[styles.inputField, globalStyles.text]}
+        id="username_placeholder"
```
Comment (Aidasvenc) [inline screens/Onboarding/OnboardingScreen.tsx:27]: Even though these are needed for the tests, they are visually polluting the code :/ unfortunately we are forced to leave it like that.

**Review by Aidasvenc — COMMENTED** [review]
Some minor changes needed in `eas.json`

**Review by Aidasvenc — APPROVED** [review]
LGTM now!

**Pull request merged**